File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.py
import sys
readline = sys.stdin.readline

m, n = map(int, readline().split())

id2Dic = dict()
name2Dic = dict()

for i in range(1, m+1):
    name = readline().rstrip()
    id2Dic[i] = name
    name2Dic[name] = i

for _ in range(n):
    check = readline().rstrip()
    if check.isdigit():
        print(id2Dic[int(check)])
    else:
        print(name2Dic[check])

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.swift
let input = readLine()!.split(separator: " ").map { Int($0)! }
let n = input[0], m = input[1]
var nameDict: [String: Int] = [:]
var numberDict: [Int: String] = [:]

for i in 1...n {
    let name = readLine()!
    nameDict[name] = i
    numberDict[i] = name
}

for _ in 0..<m {
    let name = readLine()!
    if let number = Int(name) {
        print(numberDict[number]!)
    } else {
        print(nameDict[name]!)
    }
}